pkg/cluster/bootstrap.go

```diff
@@ -576,3 +582,37 @@ func (c *Cluster) Snapshot(ctx context.Context, config *config.Control) error {
 	}
 	return c.managedDB.Snapshot(ctx, config)
 }
+
+// compareConfig verifies that the config of the joining control plane node coincides with the cluster's config
+func (c *Cluster) compareConfig() error {
+	agentClientAccessInfo, err := clientaccess.ParseAndValidateTokenForUser(c.config.JoinURL, c.config.Token, "node")
+	if err != nil {
+		return err
+	}
+	serverConfig, err := agentClientAccessInfo.Get("/v1-" + version.Program + "/config")
+	if err != nil {
+		return err
+	}
+	clusterControl := &config.Control{}
+	if err := json.Unmarshal(serverConfig, clusterControl); err != nil {
+		return err
+	}
+
+	// We are saving IPs of ClusterIPRanges and ServiceIPRanges in 4-bytes representation but json decodes in 16-byte
+	ipsTo16Bytes(c.config.ControlQuorumArgs.ClusterIPRanges)
+	ipsTo16Bytes(c.config.ControlQuorumArgs.ServiceIPRanges)
+
+	if !reflect.DeepEqual(clusterControl.ControlQuorumArgs, c.config.ControlQuorumArgs) {
+		logrus.Debugf("This is the server ControlQuorumArgs: %#v", clusterControl.ControlQuorumArgs)
+		logrus.Debugf("This is the local ControlQuorumArgs: %#v", c.config.ControlQuorumArgs)
+		return errors.New("Node can't join the cluster. Its config is different from the cluster's config. Please use the same config for all control plane nodes")
```

Comment:
In general errors should be a single statement summarizing the error, and steps to resolve if possible. Having multiple sentences doesn't parse as well to me. How about:
```suggestion
		return errors.New("Unable to join cluster due to critical configuration value mismatch.")
```

If we're going to do this, but not call out in the error which flags are out of sync, we should also document clearly which specific flags must be consistently set across nodes. Can you add a docs change?

Reply:
Good point. Will prepare that tomorrow